daemon/create.go

```diff
@@ -19,6 +19,7 @@ func (daemon *Daemon) ContainerCreate(name string, config *runconfig.Config, hos
 		return "", nil, fmt.Errorf("Config cannot be empty in order to create a container")
 	}
 
+	daemon.adaptContainerSettings(hostConfig)
```

Comment:
i think this call should be set after `verifyContainerSettings` so when the host doesnt have `memsw` support, we dont set it to x2 mem and then set to -1 after with a warning.